.github/workflows/redpanda-build.yml

```diff
@@ -12,59 +12,44 @@ on: [push, pull_request]
 
 jobs:
   build:
-    name: Build on ${{ matrix.container }}
+    name: build redpanda
     runs-on: ubuntu-20.04
-    container: ${{ matrix.container }}
-    strategy:
-      matrix:
-        container: ['ubuntu:21.04']
 
     steps:
         - name: checkout
           uses: actions/checkout@v2
 
-        - name: update container base
+        - name: prepare toolchain container
           run: |
-            export DEBIAN_FRONTEND=noninteractive
-            apt-get update -y
-            apt-get install -y git sudo pkg-config ccache
-
-        - name: install dependencies
-          run: ./install-dependencies.sh
-
-        - name: install g++-11
-          run: |
-            DEBIAN_FRONTEND=noninteractive apt install -y g++-11
-            sudo update-alternatives \
-              --install /usr/bin/gcc gcc /usr/bin/gcc-11 60 \
-              --slave /usr/bin/g++ g++ /usr/bin/g++-11
-            sudo update-alternatives --auto gcc
-
-        # there is no analog to ${{ github.workspace }} for the home directory
-        - name: prepare cache variables
-          id: prepare_cache_variables
-          run: |
-            echo ::set-output name=home::$HOME
-            echo ::set-output name=timestamp::$(date +'%Y-%m-%d-%H-%M-%S')
+            docker pull docker.io/vectorized/redpanda-toolchain || true
+            chmod 0775 install-dependencies.sh
+            cp tools/docker/Dockerfile.ignore .dockerignore
+            docker build \
+              --cache-from docker.io/vectorized/redpanda-toolchain \
+              --tag rp-toolchain \
+              --file tools/docker/Dockerfile \
+              .
+            rm .dockerignore
 
         - name: ccache cache files
           uses: actions/cache@v2
           with:
-            path: ${{ steps.prepare_cache_variables.outputs.home }}/.ccache
-            key: ${{ matrix.container }}-ccache-${{ steps.prepare_cache_variables.outputs.timestamp }}
-            restore-keys: ${{ matrix.container }}-ccache-
+            path: /dev/shm/redpanda
+            key: ccache-${{github.ref}}
```

Comment:
Will this fallback to `dev` if this fails?
```
2021-11-01T01:22:13.6661837Z Cache not found for input keys: ccache-refs/pull/2730/merge, ccache
```
Maybe it's not necessary/there's no key for dev: https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

Reply:
yeah, it'll check whether there's one for dev when an entry for the current pr's gitref is not present